src/compare_covermaps.py

```diff
@@ -214,6 +214,70 @@ def mapper(x):
     def __repr__(self) -> str:
         return self.title + " " + repr(self.countries)
 
+    def get_binary_image(self, aoi, projection="EPSG:4326"):
+        """
+        Creates a binary image for a covermap in given country.
+        """
+        binary_image = (
+            self.ee_asset.mosaic().clip(aoi).reproject(crs=projection, scale=self.resolution)
+        )
+
+        # Convert the image to binary classes: 1 (crop), 0 (noncrop)
+        if self.crop_labels is None:
+            binary_image = binary_image.gte(self.probability)
+
+        # Check if crop_labels are an ordered list
+        elif len(self.crop_labels) > 2 and self.crop_labels == list(
+            range(self.crop_labels[0], self.crop_labels[-1] + 1)
+        ):
+            binary_image = binary_image.gte(self.crop_labels[0]).And(
+                binary_image.lte(self.crop_labels[-1])
+            )
+        else:
+            binary_image = binary_image.eq(self.crop_labels[0])
+            if len(self.crop_labels) > 1:
+                for crop_value in self.crop_labels[1:]:
+                    binary_image = binary_image.Or(binary_image.eq(crop_value))
+
+        return binary_image.rename("crop")
+
+    def compute_map_area(self, country: str, projection="EPSG:4326"):
+        """
+        Calculates the area in pixels of each class in the map clipped to the country.
+        """
+        aoi = ee.FeatureCollection("FAO/GAUL/2015/level0").filter(
+            ee.Filter.eq("ADM0_NAME", country)
+        )
+
+        binary_image = self.get_binary_image(aoi=aoi, projection=projection)
+
+        crop_px_sum = (
+            binary_image.reduceRegion(
+                reducer=ee.Reducer.sum(),
+                geometry=aoi.geometry(),
+                scale=self.resolution,
+                maxPixels=1e12,
+            )
+            .get("crop")
+            .getInfo()
+        )
+
+        noncrop_px_sum = (
+            binary_image.Not()
+            .reduceRegion(
+                reducer=ee.Reducer.sum(),
+                geometry=aoi.geometry(),
+                scale=self.resolution,
+                maxPixels=1e12,
+            )
+            .get("crop")
+            .getInfo()
+        )
+
```

Comment:
Here's an alternative to getting the crop/non-crop proportion without the need for two reduceRegion calls.
It also may be more "accurate" since the proportions are computed using area vs pixel count.
```
geom = aoi.geometry()
area = geom.area()
crop_proportion = binary_image \
    .multiply(ee.Image.pixelArea()) # Multiplies each pixel by its area
    .divide(area)                   # Divides each pixel by the total area
    .reduceRegion(
        reducer=ee.Reducer.sum(), 
        geometry=geom, 
        scale=self.resolution, 
        maxPixels=1e12,
        bestEffort=True        # Should reduce scale if computation times out
    ).get("crop").getInfo()

w_j = np.array([1-crop_proportion, crop_proportion])
return w_j
```

Reply:
Can be multiplied by `area` to get `a_j`

Reply:
Never mind I see you need the exact amount of pixels for the producer variance estimator 🤦